### PR TITLE
Run jupyter notebook as non-root, added spark driver options

### PR DESCRIPTION
**Pull request by huadongliu**
Tested in norcal-devc.

### CODE REVIEW

File: paasta_tools/cli/cmds/spark_run.py [hunk 1]
```diff
@@ -165,6 +184,13 @@ def get_spark_configuration(
     spark_conf['SPARK_EXECUTOR_CORES'] = args.executor_cores
     spark_conf['SPARK_EXECUTOR_MEMORY'] = '%dg' % args.executor_memory
 
+    if args.driver_max_result_size:
+        spark_conf['SPARK_DRIVER_MAX_RESULT_SIZE'] = '%dg' % args.driver_max_result_size
```
Comment (sagar8192) [inline paasta_tools/cli/cmds/spark_run.py]: Is the 'g' in here for GB? I yes, can we mention the unit in the help message?

File: paasta_tools/cli/cmds/spark_run.py [hunk 2]
```diff
@@ -227,7 +255,7 @@ def configure_and_run_docker_container(
         return 1
     # Changes at docker ENTRYPOINT or CMD does not work.
     elif docker_cmd == 'jupyter':
-        docker_cmd = 'jupyter notebook -y --ip=%s --notebook-dir=%s --allow-root' % (
+        docker_cmd = 'jupyter notebook -y --ip=%s --notebook-dir=%s' % (
```
Comment (sagar8192) [inline paasta_tools/cli/cmds/spark_run.py]: Just out of curiosity, why did we need it before?

Comment (huadongliu) [reply paasta_tools/cli/cmds/spark_run.py]: Previously we didn't use --user for docker run. Root was the default user in the container to run jupyter.

File: paasta_tools/cli/cmds/spark_run.py (hunk 1 above)
Comment (huadongliu) [reply paasta_tools/cli/cmds/spark_run.py]: Yeah, the help message has it: 'Limit of total size in GB of serialized results of all partitions'

**Pull request merged**